mypy/types.py

```diff
@@ -783,6 +784,30 @@ def deserialize(self, data: JsonDict) -> 'EllipsisType':
         return EllipsisType()
 
 
+class TypeType(Type):
+    """For types like Type[User].
+
+    This annotates variables that are class objects, constrained by
+    the type argument.  See PEP 484 for more details.
+    """
+
+    item = None  # type: Type
```

Comment:
Add comment describing what is supported here, as this can't be an arbitrary type.